topology/nhlt/nhlt-processor.c

```diff
@@ -0,0 +1,321 @@
+/*
+  Copyright(c) 2021 Intel Corporation
+  All rights reserved.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of version 2 of the GNU General Public License as
+  published by the Free Software Foundation.
+
+  This program is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St - Fifth Floor, Boston, MA 02110-1301 USA.
+  The full GNU General Public License is included in this distribution
+  in the file called LICENSE.GPL.
+*/
+
+#include <errno.h>
+#include <stdio.h>
+#include <alsa/input.h>
+#include <alsa/output.h>
+#include <alsa/conf.h>
+#include <alsa/error.h>
+#include "pre-processor.h"
+#include "nhlt-processor.h"
+#include "nhlt.h"
+
+#define MAX_ENDPOINT_COUNT 20
+
+static int add_bytes_to_data_section(struct tplg_pre_processor *tplg_pp, char *src)
+{
+	const char *data_name = "nhlt";
+	snd_config_t *top_data_section;
+	snd_config_t *bytes;
+	snd_config_t *nhlt;
+	int ret;
+
+	/* add the nhlt bytes as new config into existing SectionData*/
+	ret = snd_config_search(tplg_pp->output_cfg, "SectionData", &top_data_section);
```

Comment:
can we use tplg_create_config_template() to create the Data config with bytes in it?